### DISCUSSION TITLE
iOS：briefcase run iOS  Pause here.

### DISCUSSION THREAD

**Discussion by jlij** [beeware/briefcase#1280] [Q&A]
Configure argc/argv...
Initializing Python runtime...
Installing Python NSLog handler...
Running app module: littlebee
---------------------------------------------------------------------------
/Users/goodtime/Library/Developer/CoreSimulator/Devices/D0F568D7-8062-4C66-B538-4D36F30E4BD7/data/Containers/Bundle/Application/02B3D6A9-3F0C-4264-B8EE-AD5343568DF5/LittleBee.app/app_packages/toga_iOS/app.py:95: DeprecationWarning: There is no current event loop
  self.loop = asyncio.get_event_loop()

**Comment by freakboy3742** [comment]
I have no idea what you're trying to communicate. Is this a question? A report of a problem? A request for assistance? 

If you want someone to help you, you need to ask a full question, and provide enough detail for them to actually help you. A stack trace with no context doesn't do that.

**Comment by jlij** [comment]
Sorry, it did not generate a problem report.
Can be successfully installed on virtual iOS, but gets stuck when entering the interface.
Made with BeeWare
The clue that can be given is the first time I sent it out.

**Reply by freakboy3742** [reply]
Again - what are you asking for? Is this a request for assistance? A report that something has gone wrong? What do you mean by "gets stuck"? 

It's literally impossible for us to help you unless you tell us what you have done, what you expected, and what is happening.

**Comment by jlij** [comment]
Compiling iOS App on macOS，Using commands “briefcase run iOS" ，Successfully installed App，
But when opening the app, it always gets stuck on the "Made with BeeWare" page.
I have tried multiple versions of toga，But the results are all the same.

**Reply by freakboy3742** [reply]
*What* iOS app? Are you doing the BeeWare Tutorial? Is it an app that you've written? What version of iOS are you using? What version of Xcode? What version of macOS?

If you don't give me enough information to reproduce the problem you're seeing, there's very little I can do to help you.

**Comment by jlij** [comment]
macOS: Big Sur 11.4
Xcode: 13.2 (13C90)
iOS:15.2
python:3.11.3
The program is used to access the database，The access database driver used is “pg8000”。

**Reply by mhsmith** [reply]
Do you have the same problem when running the BeeWare tutorial?

If not, the issue is probably with your own code. To find out exactly where, try disabling it piece by piece until the problem stops happening.

**Reply by jlij** [reply]
Successfully used on macOS.

**Reply by mhsmith** [reply]
Please answer my question and follow my advice above, otherwise it'll be impossible to help you.

**Reply by jlij** [reply]
Received, thank you for your reply.

**Reply by jlij** [reply]
According to your method, find the reason: pg8000 does not seem to support iOS.
What should I replace it with?

**Reply by freakboy3742** [reply]
Having not tried to access a Postgres database from iOS, I can't recommend an alternative.

**Reply by jlij** [reply]
Based on your experience, what database should I use?

**Reply by freakboy3742** [reply]
As a *database* I'd always recommend Postgres, unless your data set is small and local, in which case SQLite might be an option. But I haven't tried to access Postgres from iOS, so I can't recommend a DBAPI wrapper.

**Comment by jlij** [comment]
toga==0.3.1：
App finished launching.
Not implemented: TextInput.clear_error()
Not implemented: TextInput.set_on_lose_focus()
Not implemented: TextInput.set_on_gain_focus()
Not implemented: TextInput.clear_error()
Not implemented: TextInput.set_on_lose_focus()
Not implemented: TextInput.set_on_gain_focus()
Child process terminated with signal 15: Terminated

**Reply by jlij** [reply]
Is it caused by Textinput.

**Reply by freakboy3742** [reply]
In the near future, we're going to stop responding to your requests, because you are not providing any details that would help us diagnose a problem.

We have an extensive test suite that validates that TextInput works on iOS, so from our perspective, TextInput works. 

It is entirely possible that you have found a bug. However, we do not have magic powers, and we cannot read your mind. We don't know what is in your code. If you want us to help you, you need to provide us enough detail to reproduce your problem. 

This means providing *explicit* reproduction instructions - the *exact* code that reproduces a problem, and the *precise* conditions under which you are operating, the *exact* inputs you provide to generate a problem, and the *full* logs for what happens when the app crashes.

This does *not* mean that you send us the complete code for your entire custom project. It means you need to generate a reproduction case - the simplest app that demonstrates the problem you are seeing. 

Unless and until you provide these details, we are *unable to help you*.

And, to be clear - this isn't a "once off request". It applies to *every* bug report you make of the BeeWare project - and, for that matter, *any other programming project in existence*.

**Reply by jlij** [reply]
Received, thank you for your reply.

**Comment by jlij** [comment]
After investigation, it was found that it is not a problem with "pg8000" and it can obtain data normally. Guess it's the "data" issue with the DetailedList. The format of the "data" strictly follows the example you provided, and it works normally on Windows, Android, and MacOS, except for exceptions on iOS. If you give an empty "list" to a "data", it can run normally.

**Reply by jlij** [reply]
```
import toga
from toga import App, Box, DetailedList
from toga.constants import COLUMN
from toga.style import Pack

class HelloWorld(App):
    def startup(self):
        self.dl = DetailedList(style=Pack(flex=1)
        self.box = Box(
            children=[self.dl],
            style=Pack(direction=COLUMN))
        self.dl.data = [{'icon': None,
                         'title': 'John Doe',
                         'subtitle': 'Employee of the Month'}]
        self.main_window = toga.MainWindow()
        self.main_window.content = self.box
        self.main_window.show()
def main():
    return HelloWorld()
```

**Reply by mhsmith** [reply]
> it works normally on Windows, Android, and MacOS, except for exceptions on iOS

I'm not sure if you're still asking for help, but if you are, you'll have to post the exception.

**Reply by jlij** [reply]
There is no way to issue an exception report because it did not generate. The above code is the cause of the problem. After using "briefcase run iOS" and installing the app into virtual iOS, it will pause on the startup page and disconnect from macOS. After removing the code, it is found that there is a problem with "data" in "DetailedList". Adding data will cause the problem. Providing an empty list will allow it to run normally, and the data format will follow the methods provided in the tutorial, And it can run on other platforms. After disconnecting, the input on the console is:Child process terminated with signal 15: Terminated.

**Comment by jlij** [comment]
Really in need of help, can you paste the above code into the macOS environment to compile iOS, and open and run it to understand the problem I'm talking about?

**Reply by freakboy3742** [reply]
I can confirm the iOS DetailedList widget appears to have a bug at present. I'm not sure what the source of the problem is, but the DetailedList example app doesn't current run either. We're in the process of auditing and building automated tests for all the widgets; DetailedList will be looked at reasonably soon.
